--- origsrc/binutils-2.21/bfd/config.bfd	2010-12-01 05:03:10.000000000 -0600
+++ src/binutils-2.21/bfd/config.bfd	2011-06-26 22:14:43.400864800 -0500
@@ -632,7 +632,7 @@ case "${targ}" in
     targ_selvecs="bfd_elf32_i386_vec i386linux_vec i386pei_vec x86_64pei_vec bfd_elf64_l1om_vec"
     want64=true
     ;;
-  x86_64-*-mingw*)
+  x86_64-*-mingw* | x86_64-*-cygwin*)
     targ_defvec=x86_64pe_vec
     targ_selvecs="x86_64pe_vec x86_64pei_vec bfd_elf64_x86_64_vec bfd_elf64_l1om_vec i386pe_vec i386pei_vec bfd_elf32_i386_vec"
     want64=true
--- origsrc/binutils-2.21/binutils/configure.in	2010-04-22 09:37:05.000000000 -0500
+++ src/binutils-2.21/binutils/configure.in	2011-06-26 22:18:50.714116000 -0500
@@ -257,7 +257,7 @@ changequote([,])dnl
 	  BUILD_WINDRES='$(WINDRES_PROG)$(EXEEXT)'
 	  BUILD_WINDMC='$(WINDMC_PROG)$(EXEEXT)'
 	  ;;
-	x86_64-*-mingw*)
+	x86_64-*-mingw* | x86_64-*-cygwin*)
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
 	    DLLTOOL_DEFAULT="-DDLLTOOL_DEFAULT_MX86_64"
--- origsrc/binutils-2.21/ld/configure.tgt	2010-12-01 05:03:47.000000000 -0600
+++ src/binutils-2.21/ld/configure.tgt	2011-06-26 22:18:01.074829400 -0500
@@ -271,7 +271,11 @@ i[3-7]86-*-pe)		targ_emul=i386pe ;
 			targ_extra_ofiles="deffilep.o pe-dll.o" ;;
 i[3-7]86-*-cygwin*)	targ_emul=i386pe ;
 			targ_extra_ofiles="deffilep.o pe-dll.o"
-			test "$targ" != "$host" && LIB_PATH='${tooldir}/lib/w32api' ;;
+			test "$targ" != "$host" && LIB_PATH='${tooldir}/lib/w32api:=/usr/lib/w32api' ;;
+x86_64-*-cygwin*)	targ_emul=i386pep ;
+			targ_extra_emuls=i386pe
+			targ_extra_ofiles="deffilep.o pep-dll.o pe-dll.o"
+			test "$targ" != "$host" && LIB_PATH='${tooldir}/lib/w32api:=/usr/lib/w32api' ;;
 i[3-7]86-*-mingw32*)	targ_emul=i386pe ;
 			targ_extra_ofiles="deffilep.o pe-dll.o" ;;
 x86_64-*-mingw*)	targ_emul=i386pep ;
